distribution/registry.go

```diff
@@ -113,3 +119,16 @@ func digestFromManifest(m *schema1.SignedManifest, localName string) (digest.Dig
 	}
 	return manifestDigest, len(payload), nil
 }
+
+type existingTokenHandler struct {
+	token string
+}
+
+func (th *existingTokenHandler) Scheme() string {
+	return "bearer"
+}
+
+func (th *existingTokenHandler) AuthorizeRequest(req *http.Request, params map[string]string) error {
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", th.token))
```

Comment:
After some thought, we may need to restrict the token to a particular host. Otherwise, we risk credential leak upon redirect. Also, tokens should also be included _only_ on https requests, unless otherwise specified.